apps/vc-api/src/config/env-vars-validation-schema.ts

```diff
@@ -17,8 +17,40 @@
 
 import * as Joi from 'joi';
 
+export enum DB_TYPES {
+  SQLITE_INMEMORY = 'SQLITE_INMEMORY',
+  SQLITE = 'SQLITE',
+  POSTGRES = 'POSTGRES'
+}
+
 export const envVarsValidationSchema = Joi.object({
   NODE_ENV: Joi.string().valid('development', 'production', 'test').default('development'),
   PORT: Joi.number().integer().positive().default(3000),
-  BASE_URL: Joi.string().uri().default('http://localhost:3000')
-});
+  BASE_URL: Joi.string().uri().default('http://localhost:3000'),
+  DB_TYPE: Joi.string()
+    .valid(...Object.values(DB_TYPES))
+    .default(DB_TYPES.SQLITE_INMEMORY)
+})
+  .when('.DB_TYPE', {
+    switch: [
+      { is: Joi.required().valid(DB_TYPES.SQLITE), then: { SQLITE_FILE: Joi.string().required() } },
+      {
+        is: Joi.required().valid(DB_TYPES.POSTGRES),
+        then: {
+          POSTGRES_DB_HOST: Joi.string().hostname().required(),
+          POSTGRES_DB_PORT: Joi.number().port().required(),
+          POSTGRES_DB_USER: Joi.string().required(),
+          POSTGRES_DB_PASSWORD: Joi.string().required(),
+          POSTGRES_DB_NAME: Joi.string().required()
+        }
+      }
+    ]
+  })
+  .when('.DB_TYPE', {
```

Comment:
why another clause of `.when`? Can we add `is` in previouse `when.switch`?

Reply:
I needed to use this hack as setting common attributes for `is: Joi.required().valid(DB_TYPES.SQLITE, DB_TYPES.POSTGRES)` in the switch is not possible. It triggers some inconsistent behavior of `Joi`.

Maybe, repeating this twice would be more readable? What do you think?
```typescript
DB_DROP_ON_START: Joi.boolean().default(false),
DB_SYNCHRONIZE: Joi.boolean().default(false),
DB_RUN_MIGRATIONS: Joi.boolean().default(true)
```

Reply:
So, it would look like this, but we would have repeated code:
```typescript
.when('.DB_TYPE', {
    switch: [
      {
        is: Joi.required().valid(DB_TYPES.SQLITE),
        then: {
          SQLITE_FILE: Joi.string().required(),
          DB_DROP_ON_START: Joi.boolean().default(false),
          DB_SYNCHRONIZE: Joi.boolean().default(false),
          DB_RUN_MIGRATIONS: Joi.boolean().default(true)
        }
      },
      {
        is: Joi.required().valid(DB_TYPES.POSTGRES),
        then: {
          POSTGRES_DB_HOST: Joi.string().hostname().required(),
          POSTGRES_DB_PORT: Joi.number().port().required(),
          POSTGRES_DB_USER: Joi.string().required(),
          POSTGRES_DB_PASSWORD: Joi.string().required(),
          POSTGRES_DB_NAME: Joi.string().required(),
          DB_DROP_ON_START: Joi.boolean().default(false),
          DB_SYNCHRONIZE: Joi.boolean().default(false),
          DB_RUN_MIGRATIONS: Joi.boolean().default(true)
        }
      }
    ]
  })
```

Reply:
If these variables independent from `DB_TYPE` can we validate them outside of `when(DB_TYPE`?

Reply:
They are dependent on the `DB_TYPE`. They are required for Postgres and SQLite, but not for in-memory SQLite

Reply:
Then may be add `.with(DB_TYPE, DP_DROP_ON_START,..)`?

Reply:
But `.with` would require them to be defined and optional first, but in the case of in-memory SQLite, they are ignored. I think this may be confusing for someone who will make interpretations based on this schema.